src/noit_filters.c

```diff
@@ -101,10 +106,60 @@ noit_filter_initialized() {
   return initialized;
 }
 void
+noit_filter_update_last_touched(const char *filterset_name) {
+  struct timeval *now = (struct timeval *)calloc(1, sizeof(struct timeval));
+  mtev_gettimeofday(now, NULL);
+  LOCKFSLT();
+  mtev_hash_replace(filterset_last_touched, strdup(filterset_name), strlen(filterset_name), (void *)now, free, free);
```

Comment:
I am new to these implementations but can we not allocate and free this memory outside of the the critical section at least?

There's no need to duplicate the string while the lock is held, i don't know if the call to ``free`` can be done outside the lock since it seems to be embedded in the call.

Reply:
No reason.  This whole PR was scraped due to performance concerns.